File: app/src/main/java/com/aquaguard/domain/usecase/ToggleValveUseCase.java
package com.aquaguard.domain.usecase;

import com.aquaguard.domain.repository.WaterRepository;
import com.aquaguard.domain.repository.RepositoryCallback;

import javax.inject.Inject;

public class ToggleValveUseCase {
    private final WaterRepository waterRepository;

    @Inject
    public ToggleValveUseCase(WaterRepository waterRepository) {
        this.waterRepository = waterRepository;
    }

    public void execute(String deviceId, boolean open, String reason, RepositoryCallback<Void> callback) {
        waterRepository.toggleValve(deviceId, open, "USER", reason, callback);
    }
}

File: Aqua Guard/frontend/app/src/main/java/com/aquaguard/domain/usecase/ToggleValveUseCase.kt
package com.aquaguard.domain.usecase

import com.aquaguard.domain.repository.WaterRepository
import javax.inject.Inject

class ToggleValveUseCase @Inject constructor(
    private val waterRepository: WaterRepository
) {
    suspend operator fun invoke(deviceId: String, open: Boolean, reason: String): Result<Unit> {
        return waterRepository.toggleValve(deviceId, open, "USER", reason)
    }
}
